### PR TITLE
refactor: Convert broadcast to TypeScript

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: src/script/broadcast/BroadcastRepository.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  private readonly clientRepository: ClientRepository;
+  private readonly conversationRepository: ConversationRepository;
+  private readonly cryptographyRepository: CryptographyRepository;
+  private readonly logger: Logger;
+  private readonly messageSender: MessageSender;
+
+  /**
+   * @param {BroadcastService} broadcastService - Backend REST API broadcast service implementation
+   * @param {ClientRepository} clientRepository - Repository for client interactions
+   * @param {ConversationRepository} conversationRepository - Repository for conversation interactions
```
Comment (bennycode) [inline src/script/broadcast/BroadcastRepository.ts]: Types not needed in TSDoc

File: src/script/broadcast/BroadcastRepository.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    amplify.subscribe(
+      WebAppEvents.BROADCAST.SEND_MESSAGE,
+      ({genericMessage, recipients}: {genericMessage: GenericMessage; recipients: User[]}) => {
+        this.broadcastGenericMessage(genericMessage, recipients);
+      },
+    );
+  }
+
+  /**
+   * @param genericMessage Generic message that will be send
```
Comment (bennycode) [inline src/script/broadcast/BroadcastRepository.ts:89]:
```suggestion
   * @param genericMessage - Generic message that will be send
```

File: src/script/broadcast/BroadcastService.ts [hunk 3]
```diff
@@ -20,11 +20,18 @@
 import {UserClients} from '@wireapp/api-client/dist/conversation';
 import {BackendClient} from '../service/BackendClient';
 
-// Broadcast service for all broadcast calls to the backend REST API.
+export interface BroadcastPayload {
+  /** Map with per-recipient data */
```
Comment (bennycode) [inline src/script/broadcast/BroadcastService.ts]: Comment can be removed and `recipients` can be typed with: https://github.com/wireapp/wire-web-packages/blob/master/packages/api-client/src/conversation/OTRRecipients.ts

File: src/script/broadcast/BroadcastService.ts [hunk 4]
```diff
@@ -39,11 +46,9 @@ export class BroadcastService {
...
-   * @param {Object} payload.recipients - Map with per-recipient data
-   * @param {string} payload.sender - Client ID of the sender
-   * @param {Array<string>|boolean} preconditionOption - Level that backend checks for missing clients
-   * @returns {Promise} Promise that resolves when the message was sent
+   * @param payload Payload to be posted
+   * @param preconditionOption Level that backend checks for missing clients
+   * @returnsPromise that resolves when the message was sent
    */
   postBroadcastMessage(
     payload: {recipients: {}; sender: string},
```
Comment (bennycode) [inline src/script/broadcast/BroadcastService.ts:53]:
```suggestion
     payload: {recipients: OTRRecipients; sender: string},
```

File: src/script/broadcast/BroadcastRepository.ts (hunk 2 above)
Comment (ffflorian) [reply src/script/broadcast/BroadcastRepository.ts:89]: I am against this, the `-` will appear before the message when hovering over the parameter making it look like a list:

image

Comment (bennycode) [reply src/script/broadcast/BroadcastRepository.ts:89]: Ok

**Pull request merged**